test/unit/utils/params.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include <cstddef>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "ur_api.h"
+#include "ur_params.hpp"
+
+template <typename T> std::unique_ptr<T> createParams() {
+    return std::make_unique<T>();
+}
+
+template <typename T> class ParamsTest : public testing::Test {
+  protected:
+    ParamsTest() : params(createParams<T>()) {}
+    ~ParamsTest() override {}
+
+    std::unique_ptr<T> params;
```

Comment:
Can't we just store `T` directly, instead of using a unique_ptr?

Reply:
No, I need T to stay at the same address because I'm using & to an internal variable in a constructor.